### ISSUE TITLE
Update librdkafka

### ISSUE THREAD

**Issue by lukesteensen** [timberio/vector#5491]
There was a report in [discord](https://discord.com/channels/742820443487993987/746070591097798688/786881053163388959) of a strange Kafka error related to API protocol versions:

```
DEBUG librdkafka: librdkafka: PROTOERR [thrd:foo.cloudapp.net:]: my-kafka-broker:9092/1003: Protocol parse failure for ApiVersion v3 at 3/6 (rd_kafka_handle_ApiVersion:1915) (incorrect broker.version.fallback?)    
Dec 11 08:37:17.946 DEBUG librdkafka: librdkafka: PROTOERR [thrd:my-kafka-broker:]: my-kafka-broker:9092/1003: ApiArrayCnt -1 out of range
```

Our current version of `rdkafka` should work perfectly with their reported version of Kafka (2.1.1), but it's possible they're hitting a bug in the underlying library. I noticed that the latest version of `librdkafka` is 1.5.3 and our `rdkafka` crate is only on 1.4.2, with one entry in the changelog mentioning `broker.version.fallback`. It would be good to update to the lasest to ensure this isn't a bug that's been fixed upstream.

This will likely involve contributing the update to the `rdkafka` crate we use.

**Comment by jamtur01** [comment]
We're stuck on an `rdkafka` release here: https://github.com/fede1024/rust-rdkafka/issues/299

**Comment by binarylogic** [comment]
It looks like a new version has been released: https://github.com/fede1024/rust-rdkafka/issues/299#issuecomment-770200280

**Comment by eeyun** [comment]
Worth mentioning that rdkafka `0.25.0` requires a tokio runtime of `1.x` now. So this is blocked.

**Comment by pablosichert** [comment]
The bump to `rdkafka 0.26.0` has landed with #5872.
